pandas/tests/tslibs/test_parsing.py

```diff
@@ -152,25 +152,25 @@ def test_parsers_month_freq(date_str, expected):
         ("2011-12-30T00:00:00", "%Y-%m-%dT%H:%M:%S"),
         ("2011-12-30T00:00:00UTC", "%Y-%m-%dT%H:%M:%S%Z"),
         ("2011-12-30T00:00:00Z", "%Y-%m-%dT%H:%M:%S%z"),
-        ("2011-12-30T00:00:00+9", "%Y-%m-%dT%H:%M:%S%z"),
-        ("2011-12-30T00:00:00+09", "%Y-%m-%dT%H:%M:%S%z"),
+        ("2011-12-30T00:00:00+9", None),
+        ("2011-12-30T00:00:00+09", None),
         ("2011-12-30T00:00:00+090", None),
         ("2011-12-30T00:00:00+0900", "%Y-%m-%dT%H:%M:%S%z"),
         ("2011-12-30T00:00:00-0900", "%Y-%m-%dT%H:%M:%S%z"),
         ("2011-12-30T00:00:00+09:00", "%Y-%m-%dT%H:%M:%S%z"),
-        ("2011-12-30T00:00:00+09:000", "%Y-%m-%dT%H:%M:%S%z"),
-        ("2011-12-30T00:00:00+9:0", "%Y-%m-%dT%H:%M:%S%z"),
+        ("2011-12-30T00:00:00+09:000", None),
+        ("2011-12-30T00:00:00+9:0", None),
         ("2011-12-30T00:00:00+09:", None),
         ("2011-12-30T00:00:00.000000UTC", "%Y-%m-%dT%H:%M:%S.%f%Z"),
         ("2011-12-30T00:00:00.000000Z", "%Y-%m-%dT%H:%M:%S.%f%z"),
-        ("2011-12-30T00:00:00.000000+9", "%Y-%m-%dT%H:%M:%S.%f%z"),
-        ("2011-12-30T00:00:00.000000+09", "%Y-%m-%dT%H:%M:%S.%f%z"),
+        ("2011-12-30T00:00:00.000000+9", None),
+        ("2011-12-30T00:00:00.000000+09", None),
         ("2011-12-30T00:00:00.000000+090", None),
         ("2011-12-30T00:00:00.000000+0900", "%Y-%m-%dT%H:%M:%S.%f%z"),
         ("2011-12-30T00:00:00.000000-0900", "%Y-%m-%dT%H:%M:%S.%f%z"),
         ("2011-12-30T00:00:00.000000+09:00", "%Y-%m-%dT%H:%M:%S.%f%z"),
-        ("2011-12-30T00:00:00.000000+09:000", "%Y-%m-%dT%H:%M:%S.%f%z"),
-        ("2011-12-30T00:00:00.000000+9:0", "%Y-%m-%dT%H:%M:%S.%f%z"),
+        ("2011-12-30T00:00:00.000000+09:000", None),
+        ("2011-12-30T00:00:00.000000+9:0", None),
```

Comment:
Could you add a comment in the test to note that odd `+9` format shouldn't be supported by %z but dateutil.parse can support this format?